eng/ci/templates/official/jobs/build-extension-artifacts.yml

```diff
@@ -0,0 +1,113 @@
+parameters:
+  - name: ExtensionProjectName
+    type: string
+
+jobs:
+
+- job: BuildArtifacts
+  displayName: Build artifacts
+
+  templateContext:
+    outputParentDirectory: $(Build.ArtifactStagingDirectory)
+    outputs:
+    - output: pipelineArtifact
+      displayName: Publish nuget packages
+      path: $(Build.ArtifactStagingDirectory)/NugetPackages
+      artifact: NugetPackages
+
+  pool:
+    name: 1es-pool-azfunc
+    image: 1es-ubuntu-22.04
+    os: linux
+
+  variables:
+    ${{ if and( not(contains(variables['Build.SourceBranch'], '/release/')),  not(startsWith(variables['Build.SourceBranch'], 'refs/tags')) ) }}:
+      buildNumberTemp: $(Build.BuildNumber)
+    buildNumber: $[variables.buildNumberTemp]
+
+  steps:
+  - template: /eng/ci/templates/steps/install-dotnet.yml@self
+
+  - task: DotNetCoreCLI@2
+    displayName: Build project
+    inputs:
+      command: custom
+      custom: build
+      arguments: -p:BuildNumber=$(buildNumber) -c release
+      projects: |
+        extensions/**/${{ parameters.ExtensionProjectName }}.csproj
+
+  - task: EsrpCodeSigning@2
+    displayName: Sign assemblies
+    inputs:
+      ConnectedServiceName: ESRP Service-internal
+      FolderPath: extensions
+      Pattern: Microsoft.Azure.Functions.Worker.Extensions*.dll
```

Comment:
tiny nit: the pattern here includes `Extensions`, but the pattern for signing nuget package does not include that. Is that intentional?